src/test/shared/sam/debugger/samDebugConfigProvider.test.ts

```diff
@@ -209,6 +209,12 @@ describe('SamDebugConfigurationProvider', async () => {
             // No workspace folder:
             assert.deepStrictEqual(await debugConfigProvider.makeConfig(undefined, config.config), undefined)
 
+            // No launch.json (vscode will pass an empty config.request):
+            assert.deepStrictEqual(
```

Comment:
Consider stubbing out the showErrorMessage, so that the test doesn't throw up the actual message box in the experimental VS Code instance (that is the intended behavior, there's been an increase in tests showing message boxes over recent months)

Reply:
why not show the actual message box in the test instance?

Reply:
While the tests are running, the messages pile up. I'm guessing that could lead to unintended consequences if too many build up, or if you were to try interacting with them while the tests are running.

Reply:
Any interaction with the test instance has that risk, no? If rapid dialogs can interfere with other tests, we need a global `beforeTest` handler that cancels all dialogs before executing the next test.